streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -0,0 +1,182 @@
+        this.topic = topic;
+        this.keySerializer = keySerializer;
+        this.valSerializer = valSerializer;
+        this.keyDeserializer = keyDeserializer;
+        this.valDeserializer = valDeserializer;
+        this.parent = parent;
+    }
+
+    @Override
+    public KTable<K, V> filter(Predicate<K, V> predicate) {
+        String name = topology.newName(FILTER_NAME);
+        KTableProcessorSupplier<K, V, V> processorSupplier = new KTableFilter<>(predicate, false);
+        topology.addProcessor(name, processorSupplier, this.name);
+
+        return new KTableImpl<>(topology, name, processorSupplier, sourceNode, this);
+    }
+
+    @Override
+    public KTable<K, V> filterOut(final Predicate<K, V> predicate) {
+        String name = topology.newName(FILTER_NAME);
+        KTableProcessorSupplier<K, V, V> processorSupplier = new KTableFilter<>(predicate, true);
+
+        topology.addProcessor(name, processorSupplier, this.name);
+
+        return new KTableImpl<>(topology, name, processorSupplier, sourceNode, this);
+    }
+
+    @Override
+    public <V1> KTable<K, V1> mapValues(ValueMapper<V, V1> mapper) {
+        String name = topology.newName(MAPVALUES_NAME);
+        KTableProcessorSupplier<K, V, V1> processorSupplier = new KTableMapValues<>(mapper);
+
+        topology.addProcessor(name, processorSupplier, this.name);
+
+        return new KTableImpl<>(topology, name, processorSupplier, sourceNode, this);
+    }
+
+    @Override
+    public KTable<K, V> through(String topic,
+                                Serializer<K> keySerializer,
+                                Serializer<V> valSerializer,
+                                Deserializer<K> keyDeserializer,
+                                Deserializer<V> valDeserializer) {
+        String sendName = topology.newName(KStreamImpl.SINK_NAME);
+
+        topology.addSink(sendName, topic, keySerializer, valSerializer, this.name);
+
+        return topology.table(keySerializer, valSerializer, keyDeserializer, valDeserializer, topic);
+    }
+
+    @Override
+    public KTable<K, V> through(String topic) {
+        return through(topic, null, null, null, null);
+    }
+
+    @Override
+    public void to(String topic) {
+        String name = topology.newName(KStreamImpl.SINK_NAME);
+
+        topology.addSink(name, topic, this.name);
+    }
+
+    @Override
+    public void to(String topic, Serializer<K> keySerializer, Serializer<V> valSerializer) {
+        String name = topology.newName(KStreamImpl.SINK_NAME);
+
+        topology.addSink(name, topic, keySerializer, valSerializer, this.name);
+    }
+
+    @Override
+    public KStream<K, V> toStream() {
+        String name = topology.newName(TOSTREAM_NAME);
+
+        topology.addProcessor(name, new KStreamPassThrough(), this.name);
+
+        return new KStreamImpl<>(topology, name, Collections.singleton(sourceNode));
+    }
+
+    KTableValueGetterSupplier<K, V> valueGetterSupplier() {
```

Comment:
Is this function only going to be used for tests? If yes we could add some comments stating that.